### DISCUSSION TITLE
RPC Outbound error

### DISCUSSION THREAD

**Discussion by thewoodfish** [libp2p/rust-libp2p#5393] [General]
When trying to make RPC (request-response) requests i get this error:
`Io(Custom { kind: UnexpectedEof, error: Eof { name: "enum", expect: Small(1) } })`

I have no idea. I've been debugging for hours. It's not working. 
Funny thing, it worked days ago.

Please help. Help needed ASAP.

**Comment by thewoodfish** [comment]
It is because of the `swarm.behaviour_mut().kademlia.set_mode(Some(Mode::Server));` command.
Kademlia now conflics with Request-response

**Reply by dariusc93** [reply]
Are you positive that its setting the mode to server? Could you provide any additional logs?

**Comment by anilaltuner** [comment]
Is there an any update on this? It is still not working even without Kademlia

**Reply by dariusc93** [reply]
Could you try with the latest release of libp2p? Additionally, could you provide any example code to be able to reproduce the error that youre getting?
